services/worker/src/worker/job_runners/config/config_job_runner.py

```diff
@@ -48,3 +49,7 @@ def __init__(
             app_config=app_config,
             processing_step=processing_step,
         )
+
+    @classmethod
+    def get_lock_key(cls, dataset: str) -> str:
```

Comment:
I understand that it's only used in a config level job runner for now, but ... maybe we could move this method to `DatasetJobRunner`

Reply:
actually multiple jobs may use the same lock, so I'll move it to queue

Reply:
I added lock.git_rev() that doesn't depend on a job :) lmk what you think